core/src/main/java/org/apache/iceberg/BaseFileScanTask.java

```diff
@@ -100,167 +101,4 @@ public String toString() {
         .add("residual", residual())
         .toString();
   }
-
-  /**
-   * This iterator returns {@link FileScanTask} using guidance provided by split offsets.
-   */
-  @VisibleForTesting
-  static final class OffsetsAwareTargetSplitSizeScanTaskIterator implements Iterator<FileScanTask> {
-    private final List<Long> offsets;
-    private final List<Long> splitSizes;
-    private final FileScanTask parentScanTask;
-    private int sizeIdx = 0;
-
-    OffsetsAwareTargetSplitSizeScanTaskIterator(List<Long> offsetList, FileScanTask parentScanTask) {
-      this.offsets = ImmutableList.copyOf(offsetList);
-      this.parentScanTask = parentScanTask;
-      this.splitSizes = Lists.newArrayListWithCapacity(offsets.size());
-      if (offsets.size() > 0) {
```

Comment:
Is this a situation that can be encountered? And if it can, do we need to ensure that `parentScanTask` is in fact empty?

Reply:
I think sometimes it happens?  (This part is the original code, I just moved it)